homeassistant/components/nws/diagnostics.py

```diff
@@ -7,22 +7,22 @@
 from pynws import SimpleNWS
 
 from homeassistant.components.diagnostics import async_redact_data
-from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
 from homeassistant.core import HomeAssistant
 
-from .const import CONF_STATION, DOMAIN
+from . import NWSConfigEntry
+from .const import CONF_STATION
 
 CONFIG_TO_REDACT = {CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_STATION}
 OBSERVATION_TO_REDACT = {"station"}
 
 
 async def async_get_config_entry_diagnostics(
     hass: HomeAssistant,
-    config_entry: ConfigEntry,
+    config_entry: NWSConfigEntry,
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
-    nws_data: SimpleNWS = hass.data[DOMAIN][config_entry.entry_id].api
+    nws_data: SimpleNWS = config_entry.runtime_data.api
```

Comment:
```suggestion
    nws_data = config_entry.runtime_data.api
```

Typing should not be worked out automatically